### ISSUE TITLE
xpcall returning three values

### ISSUE THREAD

**Issue by Timidger** [chucklefish/rlua#53]
I found a strange issue with using `xpcall` that I can't reproduce in other Lua 5.3 interpreters. I tested this using the system lua (e.g I compiled it with `--no-default-features` on the repl and I found it through a very hard to follow bug while working on Way Cooler).

Here's the basic steps to reproduce:

Run `xpcall(function(foo) return foo end, print, "foo")` in both interpreters (note it doesn't really matter what the functions are, the first function has to return something though)

On my system lua (Lua 5.3.4) it returns `true` and `"foo"` (as expected)

On rlua it returns `true` `function` `"foo"`.

The second function it returns is the error function, you can see that by running this in the rlua interpreter:

`(function(a, b, c) b("a") end )(xpcall(function(foo) return foo end, print, "foo"))` => prints "a"

**Comment by Timidger** [comment]
Ok after digging around the source and seeing that you're doing a protected call always in order to avoid UB from longjmp's...

I can reproduce a similar result in the reference Lua interpreter by doing what rlua is _actually_ doing:

`pcall(function(foo) return xpcall(function(foo) return foo end, print, foo) end, "foo")`

It returns two booleans instead of two functions, but the result is the same: the return value is one value over than it should be.

**Comment by kyren** [comment]
Ugh, thanks for this, I'm sure I've implemented the xpcall wrapper incorrectly.  I'll try and fix this ASAP.

**Comment by kyren** [comment]
Sorry bout that, this should be fixed in 0.9.5.  I think that's at least the second time I've screwed up the pcall / xpcall wrappers, hopefully it works now.

**Comment by Timidger** [comment]
Awesome, thanks for fixing it so fast

**Comment by kyren** [comment]
Np, thanks for the report! 👍

**Comment by Timidger** [comment]
Excellent, just tested and 0.9.5 fixed my issue :)
